src/mgmt/srv_pool.c

```diff
@@ -407,8 +407,11 @@ ds_mgmt_create_pool(uuid_t pool_uuid, const char *group, char *tgt_dev,
 	if (rc == 0 && DAOS_FAIL_CHECK(DAOS_POOL_CREATE_FAIL_CORPC))
 		rc = -DER_TIMEDOUT;
 	if (rc != 0) {
+		if (!DAOS_FAIL_CHECK(DAOS_POOL_CREATE_FAIL_CORPC))
```

Comment:
Why do we not print the error when a fault has been injected?

Reply:
Maybe we should print the error message unconditionally (perhaps in a future commit). I thought it would be confusing if dss_rpc_send() actually succeeded and then print out an error message only because the code is injecting the fault after the fact.

If we print unconditinoally, do we care if for example there is a fault injection scenario (test expects to see -DER_TIMEDOUT), but the dss_rpc_send() also happens to actually fail with -DER_TIMEDOUT? No harm, or further confusion?

Reply:
> Maybe we should print the error message unconditionally (perhaps in a future commit). I thought it would be confusing if dss_rpc_send() actually succeeded and then print out an error message only because the code is injecting the fault after the fact.

If we inject a fault, I think we usually expect error messages. (In one extreme, if the D_ERROR line would segfault due to a bug, the fault injection might have caught it.) Hence, I feel the error in this case is acceptable.

> If we print unconditinoally, do we care if for example there is a fault injection scenario (test expects to see -DER_TIMEDOUT), but the dss_rpc_send() also happens to actually fail with -DER_TIMEDOUT? No harm, or further confusion?

I recall I had the same question when adding the fault injection. I first placed it before the dss_rpc_send call, but then realized that by placing it after the dss_rpc_send, I could simulate a CoRPC that succeeded on _some_ targets before hitting a timeout. That might outweigh the confusion. Any better options?

Reply:
It's a bit nit-picking though, I admit.

Reply:
I've updated the patch to make the error print unconditional